src/engine/Minus.cpp

```diff
@@ -228,3 +228,13 @@ std::unique_ptr<Operation> Minus::cloneImpl() const {
   copy->_right = _right->clone();
   return copy;
 }
+
+// _____________________________________________________________________________
+bool Minus::columnOriginatesFromGraph(const Variable& variable) const {
+  AD_CONTRACT_CHECK(getExternallyVisibleVariableColumns().contains(variable));
+  if (_left->getVariableColumnOrNullopt(variable).has_value() &&
+      _right->getVariableColumnOrNullopt(variable).has_value()) {
+    return _left->getRootOperation()->columnOriginatesFromGraph(variable);
+  }
+  return Operation::columnOriginatesFromGraph(variable);
```

Comment:
Isn't this just `_left->columnOriginatesFromGraph(var)` 
(+- checking of the existence), as the `right` subtree of a minus is not relevant for the result.

And if you have the combination of `variableIsExternallyVisibleAndOriginatesFromGraph` often times, 
you can factor it out (e.g. by splitting `originatesFromGraph` (base class, checking external visibility) and a virtual `impl` (basically what you have right now).